showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITOtelMetrics.java

```diff
@@ -0,0 +1,517 @@
+    client.echoCallable().futureCall(requestWithNoError).cancel(true);
+
+    // wait for the metrics to get uploaded
+    Thread.sleep(5000);
+
+    String filePath = "../test_data/testHttpJson_OperationCancelled_metrics.txt";
+    String attribute1 =
+        "\"attributes\":[{\"key\":\"language\",\"value\":{\"stringValue\":\"Java\"}},{\"key\":\"method_name\",\"value\":{\"stringValue\":\"google.showcase.v1beta1.Echo/Echo\"}},{\"key\":\"status\",\"value\":{\"stringValue\":\"CANCELLED\"}}]";
+
+    String[] params = {filePath, attribute1};
+    int result = verify_metrics(params);
+    Truth.assertThat(result).isEqualTo(0);
+
+    client.close();
+    client.awaitTermination(TestClientInitializer.AWAIT_TERMINATION_SECONDS, TimeUnit.SECONDS);
+  }
+
+  @Test
+  public void testGrpc_OperationCancelled_recordsMetrics() throws Exception {
+
+    generate_otelcol_config(
+        "4320",
+        "../test_data/testGrpc_OperationCancelled_metrics.txt",
+        "../test_data/testGrpc_OperationCancelled.yaml");
+    // initialize the otel-collector
+    setupOtelCollector("../test_data/testGrpc_OperationCancelled.yaml");
+
+    EchoClient client =
+        TestClientInitializer.createGrpcEchoClientOpentelemetry(
+            createOpenTelemetryTracerFactory("4320"));
+
+    EchoRequest requestWithNoError =
+        EchoRequest.newBuilder()
+            .setError(Status.newBuilder().setCode(Code.OK.ordinal()).build())
+            .build();
+
+    client.echoCallable().futureCall(requestWithNoError).cancel(true);
+
+    // wait for the metrics to get uploaded
+    Thread.sleep(5000);
+
+    String filePath = "../test_data/testGrpc_OperationCancelled_metrics.txt";
+    String attribute1 =
+        "\"attributes\":[{\"key\":\"language\",\"value\":{\"stringValue\":\"Java\"}},{\"key\":\"method_name\",\"value\":{\"stringValue\":\"Echo.Echo\"}},{\"key\":\"status\",\"value\":{\"stringValue\":\"CANCELLED\"}}]";
+
+    String[] params = {filePath, attribute1};
+    int result = verify_metrics(params);
+    Truth.assertThat(result).isEqualTo(0);
+
+    client.close();
+    client.awaitTermination(TestClientInitializer.AWAIT_TERMINATION_SECONDS, TimeUnit.SECONDS);
+  }
+
+  @Test
+  public void testHttpJson_OperationFailed_recordsMetrics() throws Exception {
+
+    generate_otelcol_config(
+        "4321",
+        "../test_data/testHttpJson_OperationFailed_metrics.txt",
+        "../test_data/testHttpJson_OperationFailed.yaml");
+
+    // initialize the otel-collector
+    setupOtelCollector("../test_data/testHttpJson_OperationFailed.yaml");
+
+    EchoClient client =
+        TestClientInitializer.createHttpJsonEchoClientOpentelemetry(
+            createOpenTelemetryTracerFactory("4321"));
+
+    EchoRequest requestWithError =
+        EchoRequest.newBuilder()
+            .setError(Status.newBuilder().setCode(Code.UNKNOWN.ordinal()).build())
+            .build();
+
+    client.echoCallable().futureCall(requestWithError).isDone();
```

Comment:
Also FYI about the echo RPCs. IIRC, they're sync calls (RPC that doesn't have the `async` suffix are sync) and you won't need to wait for `is.Done()`.